lockfile_unix.go

```diff
@@ -14,17 +14,13 @@ import (
 )
 
 func getLockFile(path string, ro bool) (Locker, error) {
-	var fd int
-	var err error
-	if ro {
-		fd, err = unix.Open(path, os.O_RDONLY, 0)
-	} else {
-		fd, err = unix.Open(path, os.O_RDWR|os.O_CREATE, unix.S_IRUSR|unix.S_IWUSR)
-	}
+	// Create and open the lock file to catch potential permission errors
+	// etc. early.
+	fd, err := unix.Open(path, os.O_RDWR|os.O_CREATE, unix.S_IRUSR|unix.S_IWUSR)
```

Comment:
(This now requires write privileges for read-only locks. Is that OK? I don’t know enough about `AdditionalImageStores` to tell.)

Reply:
That is okay, as we are only working on the FDs.

Reply:
I meant: can the read-only accessor be assumed to have the privileges to necessary to open the files read-write?

The privilege check happens at `open`, based on the provided flags; whether or not the returned FD is actually used for such operations later is secondary.

Reply:
Looking at https://github.com/containers/storage/commit/162e676330d4eceddb708e3226e958898d968f43 , the read-only open may be actually necessary.

Reply:
The way I read that, the store may be created with read-write access (which incidentally creates the lock files), and then somehow made available read-only (is that copied? or NFS-exported, with the possibility of concurrent writers which do still have read-write access?).  The read-only users can use the previously-created lock files to allocate locks [BTW, apparently even an exclusive lock does not require write access to the file descriptor?!].